airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -444,6 +445,16 @@ public ConnectionState getState(final ConnectionIdRequestBody connectionIdReques
     return connectionState;
   }
 
+  public ConnectionState setState(final ConnectionUpdateStateBody connectionUpdateStateBody) throws IOException {
+    final State state = new State();
+    state.setState(connectionUpdateStateBody.getState());
+    configRepository.updateConnectionState(connectionUpdateStateBody.getConnectionId(), state);
+    final Optional<State> newState = configRepository.getConnectionState(connectionUpdateStateBody.getConnectionId());
+    final ConnectionState connectionState = new ConnectionState().connectionId(connectionUpdateStateBody.getConnectionId());
+    newState.ifPresent(s -> connectionState.state(s.getState()));
+    return connectionState;
```

Comment:
Rather than do this, I'd rather just call the above `getState()` method, but the input types aren't the same.  Is there a JAVA way to allow for two input types, e.g. `public ConnectionState getState(final ConnectionIdRequestBody | ConnectionUpdateStateBody connectionIdRequestBody)`?

Reply:
TODO: Probably needs a test

Reply:
If not possible, could refactor the common logic to another method that just takes in the UUID, like these:

https://github.com/airbytehq/airbyte/blob/880c759cac06d5eccff2bfb2e51aac8a3c129f70/airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java#L500-L510